Improvements to visitor log

sgiehl opened the pull request:
This PR aims to fix #6176

Currently the visitor log looks like that:

![visitorlog_before](https://cloud.githubusercontent.com/assets/1579355/6652513/91178082-ca71-11e4-8b0c-f0151d22c761.png)

As the browser plugins are not relevant for every one and some users asked to have the full location information instead, I now added the possibility to see all information. I've also added a new icon for the device type. Overall it now looks like that:

![visitorlog_after](https://cloud.githubusercontent.com/assets/1579355/6652518/d4502d7c-ca71-11e4-96a1-440cef3407e7.png)

By default only the icons are displayed without any additional informationen. But it is now possible to hover the small icons to see more information:

![visitorlog_flag_hover](https://cloud.githubusercontent.com/assets/1579355/6652524/e9545810-ca71-11e4-8867-55edc67da01c.png)![visitorlog_browser_hover](https://cloud.githubusercontent.com/assets/1579355/6652522/e9527a18-ca71-11e4-991e-b6d1b0400200.png)![visitorlog_device_hover](https://cloud.githubusercontent.com/assets/1579355/6652523/e953a118-ca71-11e4-8f34-be8d30af0e5f.png)

In order to have all information available in the visitor log, I had to adjust the information available in the visitor details. They now additionally contain deviceTypeIcon, deviceBrand and deviceModel

sgiehl committed:
improved visitor log. it's now showing more information on hovering the small icons

sgiehl committed:
fixed tests

sgiehl committed:
split location data

sgiehl committed:
updated changelog

mnapoli commented on CHANGELOG.md:
```diff
@@ -5,6 +5,7 @@ This is a changelog for Piwik platform developers. All changes for our HTTP API'
 ## Piwik 2.12.0
 
 ### Breaking Changes
+* Visitor details now additionally contain: `deviceTypeIcon`, `deviceBrand` and `deviceModel`
```
Is this really a breaking change or rather an improvement?

sgiehl replied:
It's more an improvement. But I thought it would be worth mentioning in changelog and we didn't have another headline it would fit in.

mattab replied:
maybe create a new headline eg. `APIs Improvements` ?

mnapoli commented on plugins/Live/stylesheets/live.less:
```diff
@@ -240,3 +240,40 @@ a.visitor-log-visitor-profile-link {
+
+span.visitorLogIconWithDetails .details li {
+    height: 20px;
+}
```
I wouldn't put `span` in front of the class name, just in case those span turn to another HTML tag later (unless `span` is there for a specific reason?)

sgiehl replied:
Hm. good point. I'll remove it.

mnapoli commented:
That's a really cool improvement!

sgiehl committed:
improved css selectors

mattab commented:
Nice improvements!
 * +1 for the new icon for device type
 * +1 for hiding the plugin icons (too much noise!)7
 * +1 for more information (ie. full information!) to the user on request (that's really great and part of our vision to give users all the data when they need it, and not hide anything from them)
 * -1 for  "transparent" icons - can we try with the full opacity icons instead? 


BTW it's very good to improve the Visitor Log in general, because that's one of the most used and loved features in Piwik :-)

Commit:
moved changelog entry to new topic APIs Improvements

sgiehl committed:
changelog: moved changes to Piwik 2.13

sgiehl commented:
@mattab Ok. I've adjusted the changelog and moved it to new topic "API Improvements" for Piwik 2.13.
Regarding the "transparent" icons: The opacity is removed on hover, so the user get's a little feedback that something is happening. But that could be removed for sure, or should I just increase the opacity value. It currently starts at 0.5. Do you think starting at 0.7 would be better?

mattab commented:
I think it's better to remove the opacity... if we wanted to show feedback on hover...

1. maybe we could use the existing UI pattern on the Report footer icons, ie. add a grey background behind the icon. I guess it may look bad because many Visitor log icons don't have transparent background.... 
2. Another idea would be to draw a thin border on hover

sgiehl committed:
remove opacity

sgiehl commented:
Ok. I've removed the opacity. Adding border or grey background didn't look very good, cause of different icon sizes

mattab commented:
Here is feedback looking at UI tests:
 * the  "Goal conversion" icon (flag) is pushed out of display sometimes (not a regression). Can we add a linebreak before showing this icon to give it full display? eg. [this ui test](http://builds-artifacts.piwik.org/ui-tests.6176/11048.7/screenshot-diffs/singlediff.html?processed=../processed-ui-screenshots/UIIntegrationTest_visitors_visitorlog.png&expected=UIIntegrationTest_visitors_visitorlog.png&github=UIIntegrationTest_visitors_visitorlog.png) 
 * could we make the popover a bit wider by default? (general issue in piwik)

sgiehl committed:
Merge branch 'master' into 6176

sgiehl committed:
improved layout; icons are now floating if space is too small to display all in one row

sgiehl commented:
I've tried to fix the layout problem with the icons. Let's see if the ui tests look better now :)
Regarding the Popovers: Maybe we should open another issue for that, as that will break more ui tests.

mattab commented:
>  Regarding the Popovers: Maybe we should open another issue for that, as that will break more ui tests.

Sure it is: #7517

mattab commented:
Looks good to me! we'll need to update screenshot tests as the UI build failed in the PR
